src/lib/servers/auth.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from 'vscode';
```

Comment:
this is really cool! I think it would be worth thinking about bringing this to cli-lib (or a seperate pkg) so that CLI could benefit from this flow as well!

Reply:
Yeah, good call! If we end up using this, it would make a lot of sense to have it live in cli-lib.